src/api/useCourses.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import { CourseState, ApiCourseUpdate, Course } from "../types/Course";
+import Axios from "axios";
+import { baseUrl } from "./urls";
+
+
+export default function useCourses() {
+    const [{ courses, loading }, setState] = useState({ courses: [] as Course[], loading: true });
```

Comment:
könnte man auch in zwei einzelne useStates aufteilen oder ein Interface definieren was die types bestimmt. Dann braucht man auch nicht mehr sowas hackiges `course: [] as Course[]`

```typescript
interface CourseState {
 courses: Course[],
 loading: boolean,
}

useState<CourseState>{courses:[], loading:true}
```